pkgs/applications/networking/instant-messengers/signal-desktop/default.nix

```
@@ -1,6 +1,9 @@
 { hostPlatform, callPackage }: {
-  signal-desktop = if hostPlatform.system == "aarch64-linux"
-    then callPackage ./signal-desktop-aarch64.nix { }
-    else callPackage ./signal-desktop.nix { };
+  signal-desktop = {
+    aarch64-linux = callPackage ./signal-desktop-aarch64.nix { };
+    x86_64-linux = callPackage ./signal-desktop.nix { };
+    aarch64-darwin = callPackage ./signal-desktop-aarch64-darwin.nix { };
+    x86_64-darwin = callPackage ./signal-desktop-darwin.nix { };
+  }.${hostPlatform.system};
```

Wouldn't it be better to rename `signal-desktop.nix` to `signal-desktop.nix-x86_64-linux.nix` like the other files, so we can simplify it?
```suggestion
signal-desktop = callPackage ./"signal-desktop-${hostPlatform.system}.nix" { };
```

---

Agreed, a good example of this with a package I am a maintainer of is Lens.

---

Done. I also combined this with @RossComputerGuy's suggestion to change `hostPlatform` to `targetPlatform`.